examples/LidarISAM2Example.cpp

```diff
@@ -0,0 +1,518 @@
+/*
+In order for beginners to quickly understand how to use the iSAM2 algorithm, which is one of the better back-end optimization algorithms, 
+the factor graphs created in this code only contain odometry and loopback factors.
+It only needs to receive an odometry topic of type nav_msgs::Odometry with the name “/odometry” and then it will run
+*/
+
+#include <mutex>
+#include <thread>
+#include <nav_msgs/Path.h>
+#include <nav_msgs/Odometry.h>
+#include <geometry_msgs/PoseStamped.h>
+#include <pcl_conversions/pcl_conversions.h>
+#include <pcl/common/eigen.h>
+#include <pcl/kdtree/kdtree_flann.h>
+#include <ros/ros.h>
+#include <tf/transform_datatypes.h>
+#include <tf/transform_broadcaster.h>
+#include <eigen3/Eigen/Dense>
+
+#include <gtsam/geometry/Rot3.h>
+#include <gtsam/geometry/Pose3.h>
+#include <gtsam/slam/PriorFactor.h>
+#include <gtsam/slam/BetweenFactor.h>
+#include <gtsam/nonlinear/NonlinearFactorGraph.h>
+#include <gtsam/nonlinear/LevenbergMarquardtOptimizer.h>
+#include <gtsam/nonlinear/Marginals.h>
+#include <gtsam/nonlinear/Values.h>
+#include <gtsam/nonlinear/ISAM2.h>
+#include <gtsam/linear/NoiseModel.h>
+
+using namespace gtsam;
+
+//The definition of point cloud in LIO-SAM is used here, xyz,intensity,roll,pitch,yaw,time, contains most of the basic information.
+struct PointXYZIRPYT
```

Comment:
Could you add conversions to (as constructors) and from (as methods) in this struct? It will make the code below easier and more compact.

Reply:
Yes, adding the constructor will make the structure more complete, I've made the changes, and I've also added support for PointTypePose to be constructed from the Pose3 type, to make it easier to write to the variable.